libflashrom.c

```diff
@@ -103,429 +106,455 @@ int print(const enum flashrom_log_level level, const char *const fmt, ...)
+					case FLASHROM_FLAG_FORCE_BOARDMISMATCH:	flashctx->flags.force_boardmismatch = value; break;
+					case FLASHROM_FLAG_VERIFY_AFTER_WRITE:	flashctx->flags.verify_after_write = value; break;
+					case FLASHROM_FLAG_VERIFY_WHOLE_CHIP:	flashctx->flags.verify_whole_chip = value; break;
+				}
+			}
 
-	free(layout);
-}
+			/**
+			* @brief Return the current value of a flag in the given flash context.
+			*
+			* @param flashctx Flash context to read from.
+			* @param flag	   Flag to be read.
+			* @return Current value of the flag.
+			*/
+			bool flashrom_flag_get(const struct flashrom_flashctx *const flashctx, const enum flashrom_flag flag)
+			{
+				switch (flag) {
+					case FLASHROM_FLAG_FORCE:		return flashctx->flags.force;
+					case FLASHROM_FLAG_FORCE_BOARDMISMATCH:	return flashctx->flags.force_boardmismatch;
+					case FLASHROM_FLAG_VERIFY_AFTER_WRITE:	return flashctx->flags.verify_after_write;
+					case FLASHROM_FLAG_VERIFY_WHOLE_CHIP:	return flashctx->flags.verify_whole_chip;
+					default:				return false;
+				}
+			}
 
-/**
- * @brief Set the active layout for a flash context.
- *
- * Note: This just sets a pointer. The caller must not release the layout
- *       as long as he uses it through the given flash context.
- *
- * @param flashctx Flash context whose layout will be set.
- * @param layout   Layout to bet set.
- */
-void flashrom_layout_set(struct flashrom_flashctx *const flashctx, const struct flashrom_layout *const layout)
-{
-	flashctx->layout = layout;
-}
+			/** @} */ /* end flashrom-flash */
+
+
+
+			/**
+			* @defgroup flashrom-layout Layout handling
+			* @{
+			*/
+
+			/**
+			* @brief Mark given region as included.
+			*
+			* @param layout The layout to alter.
+			* @param name   The name of the region to include.
+			*
+			* @return 0 on success,
+			*         1 if the given name can't be found.
+			*/
+			int flashrom_layout_include_region(struct flashrom_layout *const layout, const char *name)
+			{
+				size_t i;
+				for (i = 0; i < layout->num_entries; ++i) {
+					if (!strcmp(layout->entries[i].name, name)) {
+						layout->entries[i].included = true;
+						return 0;
+					}
+				}
+				return 1;
+			}
 
-/** @} */ /* end flashrom-layout */
+			/**
+			* @brief Read a layout from the Intel ICH descriptor in the flash.
+			*
+			* Optionally verify that the layout matches the one in the given
+			* descriptor dump.
+			*
+			* @param[out] layout Points to a struct flashrom_layout pointer that
+			*                    gets set if the descriptor is read and parsed
+			*                    successfully.
+			* @param[in] flashctx Flash context to read the descriptor from flash.
+			* @param[in] dump     The descriptor dump to compare to or NULL.
+			* @param[in] len      The length of the descriptor dump.
+			*
+			* @return 0 on success,
+			*         6 if descriptor parsing isn't implemented for the host,
+			*         5 if the descriptors don't match,
```

Comment:
`descriptors don't match` is unclear. Looking at the code, I would say that the number of regions do not match, not the descriptors.